netbox/extras/models/customfields.py

```diff
@@ -627,3 +631,46 @@ def validate(self, value):
 
         elif self.required:
             raise ValidationError("Required field cannot be empty.")
+
+
+class CustomFieldChoiceSet(ChangeLoggedModel):
+    """
+    Represents a set of choices available for choice and multi-choice custom fields.
+    """
+    name = models.CharField(
+        max_length=100,
+        unique=True
+    )
+    description = models.CharField(
+        max_length=200,
+        blank=True
+    )
+    extra_choices = ArrayField(
+        base_field=models.CharField(max_length=100),
+        help_text=_('Comma-separated list of available choices (for selection fields)')
```

Comment:
Should this be new-line instead of CSV?  In the form it mentions new-line.  DB it is stored as json however.

```suggestion
        help_text=_('Line-separated list of available choices (for selection fields)')
```